sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_safety_evaluation/_red_team_agent.py

```diff
@@ -0,0 +1,1138 @@
+    DIFFICULT = "difficult"
+    AnsiAttack = "ansi_attack"
+    AsciiArt = "ascii_art"
+    AsciiSmuggler = "ascii_smuggler"
+    Atbash = "atbash"
+    Base64 = "base64"
+    Binary = "binary"
+    Caesar = "caesar"
+    CharacterSpace = "character_space"
+    CharSwap = "char_swap"
+    Diacritic = "diacritic"
+    Flip = "flip"
+    Leetspeak = "leetspeak"
+    MaliciousQuestion = "malicious_question"
+    Math = "math"
+    Morse = "morse"
+    Persuasion = "persuasion"
+    ROT13 = "rot13"
+    RepeatToken = "repeat_token"
+    SuffixAppend = "suffix_append"
+    StringJoin = "string_join"
+    Tense = "tense"
+    Tone = "tone"
+    Translation = "translation"
+    UnicodeConfusable = "unicode_confusable"
+    UnicodeSubstitution = "unicode_substitution"
+    Url = "url"
+    Variation = "variation"
+    Baseline = "baseline"
+
+    @classmethod
+    def Compose(cls, items: List["AttackStrategy"]) -> List["AttackStrategy"]:
+        for item in items:
+            if not isinstance(item, cls):
+                raise ValueError("All items must be instances of AttackStrategy")
+        if len(items) > 2: 
+            raise ValueError("Composed strategies must have at most 2 items")
+        return items
+    
+def _setup_logger():
+    """Configure and return a logger instance for the RedTeamAgent.
+
+    :return: The logger instance.
+    :rtype: logging.Logger
+    """
+    log_filename = datetime.now().strftime("%Y_%m_%d__%H_%M.log")
+    logger = logging.getLogger("RedTeamAgentLogger")
+    logger.setLevel(logging.DEBUG)  
+    file_handler = logging.FileHandler(log_filename)
+    file_handler.setLevel(logging.DEBUG)
+    formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+    file_handler.setFormatter(formatter)
+    logger.addHandler(file_handler)
+
+    return logger
+
+@experimental
+class RedTeamAgent():
+    def __init__(self, azure_ai_project, credential):
+        validate_azure_ai_project(azure_ai_project)
+        self.azure_ai_project = AzureAIProject(**azure_ai_project)
+        self.credential=credential
+        self.logger = _setup_logger()
+        self.token_manager = ManagedIdentityAPITokenManager(
+            token_scope=TokenScope.DEFAULT_AZURE_MANAGEMENT,
+            logger=logging.getLogger("RedTeamAgentLogger"),
+            credential=cast(TokenCredential, credential),
+        )
+
+        self.rai_client = RAIClient(azure_ai_project=self.azure_ai_project, token_manager=self.token_manager)
+        self.generated_rai_client = GeneratedRAIClient(azure_ai_project=self.azure_ai_project, token_manager=self.token_manager.get_aad_credential())
+
+        self.adversarial_template_handler = AdversarialTemplateHandler(
+            azure_ai_project=self.azure_ai_project, rai_client=self.rai_client
+        )
+
+        # Initialize a cache for attack objectives by risk category and strategy
+        self.attack_objectives_cache = {}
+        
+        initialize_pyrit(memory_db_type=DUCK_DB)
```

Comment:
I'm wondering if you'd be better off with `IN_MEMORY` since you're not planning to reuse this DB beyond a single run. Occasionally, we find ourselves having to delete the DB if the schema changes, for example. Using a persistent DB like this would mean users have to manually do that. With in-memory, it goes away after every run.